realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -358,15 +361,54 @@ public DynamicRealmObject getObject(String fieldName) {
     }
 
     /**
-     * Returns the {@link RealmList} of values being linked from the given field.
+     * Returns the {@link RealmList} containing only primitive values.
+     *
+     * <p>
+     * If the list contains references to other Realm objects, use {@link #getList(String)} instead.
      *
      * @param fieldName the name of the field.
+     * @param elementType the type of elements in the list. Only primitive types are supported.
      * @return the {@link RealmList} data for this field.
-     * @throws IllegalArgumentException if field name doesn't exist or it doesn't contain a list of values.
+     * @throws IllegalArgumentException if field name doesn't exist or it doesn't contain a list of primitive objects.
      */
-    public <T> RealmList<T> getValueList(String fieldName, Class<T> valueClass) {
-        // TODO implement this
-        return null;
+    public <E> RealmList<E> getList(String fieldName, Class<E> elementType) {
```

Comment:
I think https://github.com/realm/realm-java/pull/5329#discussion_r142131879 applies here as well (that `primitiveType` is a better name)

Reply:
Why `primitive` is better? It's not a primitive type at least in Java.

Reply:
Technically because the feature is called "array of primitives".

The term `element` does not portray that this is not meant to be used for explicit RealmModel classes.

Reply:
"array of primitives" is just a internal feature name and I don't think we are using it in our documentation.

Reply:
The problem is that no terminology is 100% correct without also being too vague:

`elementType`: Correct, but doesn't describe the difference between the RealmModel variant and this very well.
`valueType`: `Date` and byte[] isn't value types. Nor may others be in the future
`primitiveType` `Date/byte[]` isn't primitive types. Technically neither are any of the boxed types.

`valueType` is probably the one with the least amount of "wrongness", but IMO it is also a pretty specialized concept which I'm guessing not a lot know about. Primitive types is hopefylly someone even starting out with Java knows about.

Hence `primitiveType`

Reply:
Damn, writing this up, made me doubt if perhaps `valueType` isn't the correct word after all 🤕

Reply:
@cmelchior I used to stand behind `value` but I'm pretty sure the core name and the feature name and the way people refer to it has been `list of primitives` for a long time.

If you check the Core issue https://github.com/realm/realm-core/issues/1881 from which there are many linked issues, they are all called "list of primitives" and `value` is never there.

Reply:
We don't need to use the same word with core.
The most important thing is what is the most natural for Java developers.

Reply:
> We don't need to use the same word with core.

You don't *need to*, but I think the other alternatives allow more confusion.

Reply:
> We don't need to use the same word with core.The most important thing is what is the most natural for Java developers.

True, but what is that? :)

Reply:
Especially, if there isn't a _clear_ benefit_ to choosing another word, we should probably go with what is otherwise used in Core/Bindings/Issues.

Reply:
@cmelchior  All candidates don't describe the difference between the RealmModel variant and expected ones well. 
I could't find the reason to choose more incorrect one...

But I can accept your decision.

Reply:
I think it ended up staying `elementType`.